cmake/modules/RootMacros.cmake

```diff
@@ -283,115 +283,226 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
     set(libprefix "")
   endif()
 
-  #---Get the list of include directories------------------
-  get_directory_property(incdirs INCLUDE_DIRECTORIES)
-  # rootcling invoked on foo.h should find foo.h in the current source dir,
-  # no matter what.
-  list(APPEND incdirs ${CMAKE_CURRENT_SOURCE_DIR})
+  if((CMAKE_PROJECT_NAME STREQUAL ROOT) AND (TARGET ${ARG_MODULE}))
+    set(incdirs)
+
+    if (cxxmodules OR runtime_cxxmodules)
+       # Comments from Vassil:
+       # FIXME: We prepend ROOTSYS/include because if we have built a module
+       # and try to resolve the 'same' header from a different location we will
+       # get a redefinition error.
+       # We should remove these lines when the fallback include is removed. Then
+       # we will need a module.modulemap file per `inc` directory.
+       list(APPEND incdirs ${CMAKE_BINARY_DIR}/include)
+    else()
+       # Comments from Sergey: because of TVirtualObject.h, appearing in each dictionary
```

Comment:
Could you elaborate? Do you mean that each dictionary contains a `#include <TVirtualObject.h>`?

Reply:
Yes, each dictionary includes TVirtualObject.h, which is in io/io/inc directory. 
We could move it in core lib

Reply:
That’d make sense to me. @pcanal?

Reply:
> Comments from Sergey: because of TVirtualObject.h, appearing in each dictionary

This is not 'precise'.  TVirtualObject.h will be included in a dictionary **if and only if** it has an I/O customization rule.   (in ROOT that's 10 dicts out of more than 100 and none in core ;) ).

That said, this is a header only class and indeed it could eventually be used in a dict in core.  I would prefer if that change (moving TVirtualObject to core) had its own PR.  thanks,